DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+// Name registrar using deposits
+
+contract Deed {
+    // The Deed is a contract intended simply to hold ether
+    // It can be controlled only by the registrar and can only send ether back to the owner
+    address public owner;
+    address public bidder;
+    Registrar public registrar;
+    bytes32 public sealedBid;
+    bytes32 public hashedName;
```

Comment:
So, a label is a part of a name - in 'foo.bar.com', 'foo', 'bar', and 'com' are labels. In ENS I've been using label to refer to the hash of the actual label, since raw labels aren't handled anywhere in ENS.

Since a registrar will only register names on top of a single TLD, passing in the whole name seems unnecessary - just the one component being registered will suffice, and ensure people don't try and register names that the registrar isn't authoritative for.